management/backup.py

```diff
@@ -276,11 +280,11 @@ def perform_backup(full_backup):
 		return
 
 	# On the first run, always do a full backup. Incremental
-	# will fail. Otherwise do a full backup when the size of
-	# the increments since the most recent full backup are
-	# large.
+	# will fail. Otherwise do a full backup on weekends when
+	# the size of the increments since the most recent full
+	# backup are large.
 	try:
-		full_backup = full_backup or should_force_full(config, env)
+		full_backup = (full_backup and weekend) or should_force_full(config, env)
```

Comment:
I think @kiekerjan is right, so it should probably be:

```suggestion
		full_backup = full_backup or (weekend and should_force_full(config, env))
```

By doing it `weekend and ...`, if `weekend` is false it won't even check if it should (via `should_force_full`). `should_force_full` scans the existing backup to see it's state and may be somewhat slow, so avoiding it on non-weekend days would be nice.

Reply:
The thing is that should_force_full also checks if there are no full backups. If in that case weekend = False, the backup will fail, as an incremental backup will be tried.

Reply:
Ok, I made changes to `should_force_full` instead.